src/eduid_common/api/utils.py

```diff
@@ -92,24 +93,19 @@ def get_user(userdb, user_class=ProofingUser):
         raise ApiException('Not authorized', status_code=401)
 
 
-def get_dashboard_user():
-    return get_user(current_app.dashboard_userdb, user_class=DashboardUser)
-
-
-def save_dashboard_user(user):
+def save_and_sync_user(user):
     """
-    Save (new) user objects to the dashboard db in the new format,
-    and propagate the changes to the central user db.
+    Save (new) user object to the private userdb and propagate the changes to the central user db.
 
     May raise UserOutOfSync exception
 
     :param user: the modified user
-    :type user: eduid_userdb.dashboard.user.DashboardUser
+    :type user: current_app.private_userdb.UserClass
     """
-    if isinstance(user, User) and not isinstance(user, DashboardUser):
-        # turn it into a DashboardUser before saving it in the dashboard private db
-        user = DashboardUser(data = user.to_dict())
-    current_app.dashboard_userdb.save(user)
+    if not isinstance(user, current_app.private_userdb.UserClass):
+        # turn it into a private userdb user class before saving it in the private db
+        user = current_app.private_userdb.UserClass(data=user.to_dict())
```

Comment:
I would prefer it if we had to be a little more explicit about casting users from User to SomeOtherUser to make it easier to not save the wrong user. In effect, I would prefer if this save function would raise an exception if it was not passed a current_app.private_userdb.UserClass instance.

Reply:
Agreed.